File: HRMS/HRMS.Web/Modules/Operations/LeavePolicy/LeavePolicyForm.cs
using Serenity.ComponentModel;

namespace HRMS.Operations.Forms;

[FormScript("Operations.LeavePolicy")]
[BasedOnRow(typeof(LeavePolicyRow), CheckNames = true)]
public class LeavePolicyForm
{
    public string PolicyName { get; set; }
    public DateTime ApplicableFromDate { get; set; }
    public string Branch { get; set; }
    public int DepartmentId { get; set; }
    public int MaxConsecutiveLeavesAllowed { get; set; }
    public bool NoticePeriodLeaveAllowed { get; set; }
    public bool ProbationLeaveAllowed { get; set; }
    public ApprovalLevelMode ApprovalLevels { get; set; }
    public bool HROverridePermission { get; set; }
    public int PayrollCutoffDay { get; set; }
    public RecordStatus Status { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/LeavePolicyForm.ts
import { BooleanEditor, DateEditor, EnumEditor, initFormType, IntegerEditor, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";
import { ApprovalLevelMode } from "./ApprovalLevelMode";
import { RecordStatus } from "./RecordStatus";

export interface LeavePolicyForm {
    PolicyName: StringEditor;
    ApplicableFromDate: DateEditor;
    Branch: StringEditor;
    DepartmentId: LookupEditor;
    MaxConsecutiveLeavesAllowed: IntegerEditor;
    NoticePeriodLeaveAllowed: BooleanEditor;
    ProbationLeaveAllowed: BooleanEditor;
    ApprovalLevels: EnumEditor;
    HROverridePermission: BooleanEditor;
    PayrollCutoffDay: IntegerEditor;
    Status: EnumEditor;
}

export class LeavePolicyForm extends PrefixedContext {
    static readonly formKey = 'Operations.LeavePolicy';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!LeavePolicyForm.init) {
            LeavePolicyForm.init = true;

            var w0 = StringEditor;
            var w1 = DateEditor;
            var w2 = LookupEditor;
            var w3 = IntegerEditor;
            var w4 = BooleanEditor;
            var w5 = EnumEditor;

            initFormType(LeavePolicyForm, [
                'PolicyName', w0,
                'ApplicableFromDate', w1,
                'Branch', w0,
                'DepartmentId', w2,
                'MaxConsecutiveLeavesAllowed', w3,
                'NoticePeriodLeaveAllowed', w4,
                'ProbationLeaveAllowed', w4,
                'ApprovalLevels', w5,
                'HROverridePermission', w4,
                'PayrollCutoffDay', w3,
                'Status', w5
            ]);
        }
    }
}

[ApprovalLevelMode, RecordStatus]; // referenced types